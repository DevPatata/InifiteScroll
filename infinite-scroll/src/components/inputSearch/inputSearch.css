.input-search{
  border-top-left-radius: 5px;
  border-bottom-left-radius: 5px;
}

.input-search-button{
  border: solid 1px var(--color-grey);
  padding: 0.5rem;
}

.input-button{
  border-top-right-radius: 5px;
  border-bottom-right-radius: 5px;
}

.input-search:focus{
  outline: none;
}

.input-button:hover{
  cursor: pointer;
  color: #fff;
  background-color: var(--color-grey);
}

.form-padding{
  padding: 1rem 0;
}



