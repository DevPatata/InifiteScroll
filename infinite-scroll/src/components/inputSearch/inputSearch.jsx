import React from 'react';
import { useForm } from '../../hooks'
import './inputSearch.css'

export const InputSearch = ({ action, placeholder = 'Buscar por titulo' }) => {

  const { searchText, onInputChange } = useForm({ searchText: '' })

  const formatInputValue = (inputValue) => {
    return inputValue.toLowerCase().replace(/(^\w{1})|(\s+\w{1})/g, letra => letra.toUpperCase());
  }

  const onSearchSubmit = (event) => {
    event.preventDefault();
    action(formatInputValue(searchText));
  }

  return (
    <form onSubmit={onSearchSubmit} className='form-padding' aria-label="form">
      <input
        type='Text'
        placeholder={placeholder}
        name='searchText'
        autoComplete='off'
        className='input-search-button input-search'
        value={searchText}
        onChange={onInputChange}>
      </input>
      <button className='input-search-button input-button'>
        <i className="fas fa-search search__icon" id="icon"></i>
      </button>
    </form>
  )
}
