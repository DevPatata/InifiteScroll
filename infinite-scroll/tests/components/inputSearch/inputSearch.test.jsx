import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react';
import { InputSearch } from '../../../src/components/inputSearch/inputSearch';

describe('Prueba componente InputSearch', () => {

  test('Prueba match snapshot', () => {
    const { container } = render(<InputSearch />)

    expect(container).toMatchSnapshot();
  });

  test('Prueba cambio texto input', () => {
    render(<InputSearch />);

    const input = screen.getByRole('textbox');

    fireEvent.input(input, { target: { value: 'unTEST' } });

    expect(input.value).toBe('unTEST');
  });

  test('Prueba llamar formatInputValue cuando cambian el input', () => {
    const actionInput = jest.fn();

    render(<InputSearch action={ actionInput }/>);

    const input = screen.getByRole('textbox');
    const form = screen.getByRole('form');

    fireEvent.input(input, { target: { value: 'unTEST' } });
    fireEvent.submit(form);

    expect( actionInput ).toHaveBeenCalled();
    expect( actionInput ).toHaveBeenCalledWith('Untest');
  })
})