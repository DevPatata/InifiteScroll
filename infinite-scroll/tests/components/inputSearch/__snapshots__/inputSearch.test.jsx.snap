// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba componente InputSearch Prueba match snapshot 1`] = `
<div>
  <form
    aria-label="form"
    class="form-padding"
  >
    <input
      autocomplete="off"
      class="input-search-button input-search"
      name="searchText"
      placeholder="Buscar por titulo"
      type="Text"
      value=""
    />
    <button
      class="input-search-button input-button"
    >
      <i
        class="fas fa-search search__icon"
        id="icon"
      />
    </button>
  </form>
</div>
`;
